checks.d/marathon.py

```diff
@@ -20,98 +28,37 @@ def check(self, instance):
                 if versions_reply is not None:
                     self.gauge('marathon.versions', len(versions_reply['versions']), tags=tags)
 
-    def get_v2_apps(self, url, timeout):
-        # Use a hash of the URL as an aggregation key
-        aggregation_key = md5(url).hexdigest()
-        tags = ['url:%s' % url]
-        msg = None
-        status = None
+    def get_json(self, url, timeout):
         try:
-            r = requests.get(url + "/v2/apps", timeout=timeout)
-            if r.status_code != 200:
-                self.status_code_event(url, r, aggregation_key)
-                status = AgentCheck.CRITICAL
-                msg = "Got %s when hitting %s" % (r.status_code, url)
+            r = requests.get(url, timeout=timeout)
         except requests.exceptions.Timeout:
             # If there's a timeout
-            self.timeout_event(url, timeout, aggregation_key)
-            msg = "%s seconds timeout when hitting %s" % (timeout, url)
-            status = AgentCheck.CRITICAL
-        except Exception as e:
-            msg = e.message
-            status = AgentCheck.CRITICAL
-        finally:
-            if status is AgentCheck.CRITICAL:
-                self.service_check(self.SERVICE_CHECK_NAME, status, tags=tags, message=msg)
-                raise Exception(msg)
-
-        req_json = r.json()
-        app_count = len(req_json['apps'])
-
-        if app_count is 0:
-            status = AgentCheck.WARN
-            msg = "No marathon applications detected at %s" % url
-        else:
-            instance_count = 0
-            for app in req_json['apps']:
-                instance_count += app['instances']
-            status = AgentCheck.CRITICAL if instance_count is 0 else AgentCheck.OK
-            msg = "%s Marathon app(s) detected with %s instances running at %s" % (app_count, instance_count, url)
-
-
-        self.service_check(self.SERVICE_CHECK_NAME, status, tags=tags, message=msg)
-
-        return req_json
-
-    def get_v2_app_versions(self, url, app_id, timeout):
-        # Use a hash of the URL as an aggregation key
-        aggregation_key = md5(url).hexdigest()
-
-        try:
-            r = requests.get(url + "/v2/apps/" + app_id + "/versions", timeout=timeout)
-        except requests.exceptions.Timeout:
-            # If there's a timeout
-            self.timeout_event(url, timeout, aggregation_key)
-            self.warning("Timeout when hitting %s" % url)
-            return None
+            self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL,
+                message='%s timed out after %s seconds.' % (url, timeout),
+                tags = ["url:{}".format(url)])
+            raise Exception("Timeout when hitting %s" % url)
 
         if r.status_code != 200:
-            self.status_code_event(url, r, aggregation_key)
-            self.warning("Got %s when hitting %s" % (r.status_code, url))
+            self.status_code_event(url, r)
```

Comment:
This function does not exist anymore?
You may want to use `r.raise_for_status()` in the
```
try: 
     r = requests.get()
     r.raise_for_status()
except timeout:
    <service_check_criticial>
except httperror:
    <service_check_criticial>
else:
    <service_check_ok>`
```
above (don't know for marathon, but other 2XX status may be OK) and you can get rid of this extra if block.